mscore/pianoroll/pianoroll.cpp

```diff
@@ -56,118 +56,266 @@ PianorollEditor::PianorollEditor(QWidget* parent)
+            { QStringLiteral(":/data/icons/note-16.svg"), -4, false },
+            { QStringLiteral(":/data/icons/note-32.svg"), -5, false },
+            { QStringLiteral(":/data/icons/note-64.svg"), -6, false },
+            { QStringLiteral(":/data/icons/note-128.svg"), -7, false },
+            { QStringLiteral(":/data/icons/note-256.svg"), -8, false },
+            { QStringLiteral(":/data/icons/note-512.svg"), -9, false },
+            { QStringLiteral(":/data/icons/note-1024.svg"), -10, false },
+            { "", 0, false },
+            };
+
+      QToolBar* tbNoteLen = addToolBar("Toolbar Note Length");
+      QButtonGroup* bngrpNoteLen = new QButtonGroup();
+
+      for (LenIconData* p = _iconData; !p->_icon.isEmpty(); ++p) {
+            QToolButton* bnLen = new QToolButton();
+            QIcon icon;
+            icon.addFile(p->_icon, QSize(), QIcon::Normal, QIcon::Off);
+            bnLen->setIcon(icon);
+            bnLen->setCheckable(true);
+            int length = p->_measureFrac;
+            connect(bnLen, &QToolButton::clicked, this, [=](){this->setEditNoteLength(length);});
+
+            if (p->_selected)
+                  bnLen->setChecked(true);
+            bngrpNoteLen->addButton(bnLen);
+            tbNoteLen->addWidget(bnLen);
+            }
+
+      //----
+
+      QToolBar* tbDots = addToolBar("Toolbar Dots");
+      QButtonGroup* bngrpNoteDot = new QButtonGroup();
+
+      struct DotIconData
+      {
+          QString _icon;
+          int _len;
+          bool _selected;
+      };
+      DotIconData _iconDotData[] = {
+            { QStringLiteral(":/data/icons/note-dot.svg"), 1, false },
+            { QStringLiteral(":/data/icons/note-double-dot.svg"), 2, false },
+            { QStringLiteral(":/data/icons/note-dot3.svg"), 3, false },
+            { QStringLiteral(":/data/icons/note-dot4.svg"), 4, false },
+            { "", -1, false },
+            };
+
+
+      for (DotIconData* p = _iconDotData; p->_len != -1; ++p) {
+            QToolButton* bn = new QToolButton();
+            QIcon icon;
+            icon.addFile(p->_icon, QSize(), QIcon::Normal, QIcon::Off);
+            bn->setIcon(icon);
+            bn->setCheckable(true);
+            int length = p->_len;
+            connect(bn, &QToolButton::clicked, this, [=](){this->setEditNoteDots(length, bn);});
+
+            if (p->_selected)
+                  bn->setChecked(true);
+            bngrpNoteDot->addButton(bn);
+            tbDots->addWidget(bn);
+            }
+
+      //----
+
+      QToolBar* tbTool = addToolBar("Toolbar Edit Tool");
+      bngrpNoteLen = new QButtonGroup();
 
-      tb = addToolBar("Toolbar 2");
+      struct ToolIconData
+      {
+          QString _icon;
+          PianoRollEditTool _tool;
+          QString _tooltip;
+          bool _selected;
+      };
+      ToolIconData _iconDataTool[] = {
+            { QStringLiteral(":/data/icons/preEdit-select.svg"), PianoRollEditTool::SELECT, tr("Select Notes"), true },
```

Comment:
Not sure, but I believe it needs to be `QT_TRANSLATE_NOOP("context","string")` here in the intiializer and `qApp->translate("context",p->_tootip)` later?

Reply:
I'm not sure this is true.  I would like to get confirmation on this.